dataland-backend/src/main/kotlin/org/dataland/datalandbackend/service/InMemoryDataStore.kt

```diff
@@ -1,29 +1,91 @@
 package org.dataland.datalandbackend.service
 
 import org.dataland.datalandbackend.interfaces.DataStoreInterface
-import org.dataland.datalandbackend.model.DataSet
+import org.dataland.datalandbackend.model.CompanyMetaInformation
+import org.dataland.datalandbackend.model.DataIdentifier
 import org.dataland.datalandbackend.model.DataSetMetaInformation
+import org.dataland.datalandbackend.model.StoredCompany
+import org.dataland.datalandbackend.model.StoredDataSet
 import org.springframework.stereotype.Component
 
 /**
  * Simple implementation of a data store using in memory storage
  */
 @Component("DefaultStore")
 class InMemoryDataStore : DataStoreInterface {
-    var data = mutableMapOf<Int, DataSet>()
-    private var counter = 0
+    var data = mutableMapOf<String, StoredDataSet>()
+    private var dataCounter = 0
+    var companyData = mutableMapOf<String, StoredCompany>()
+    private var companyCounter = 0
 
-    override fun addDataSet(dataSet: DataSet): DataSetMetaInformation {
-        counter++
-        data[counter] = dataSet
-        return DataSetMetaInformation(name = dataSet.name, id = counter.toString())
+    override fun addDataSet(storedDataSet: StoredDataSet): String {
+        if (companyData.containsKey(storedDataSet.companyId)) {
+            dataCounter++
+            this.data["$dataCounter"] =
+                StoredDataSet(
```

Comment:
Warum machen wir hier eine Kopie vom storedDataSet? Warum benutzen wir es nicht einfach?
(also: Warum nicht `this.data["$dataCounter"] = storedDataSet`)

Reply:
Fixed.  Gut gesehen 👍

Reply:
ok